src/herder/PendingEnvelopes.cpp

```diff
@@ -674,10 +677,20 @@ PendingEnvelopes::eraseBelow(uint64 slotIndex)
 
     // 0 is special mark for data that we do not know the slot index
     // it is used for state loaded from database
-    mTxSetCache.erase_if([&](TxSetFramCacheItem const& i) {
+    auto erased = mTxSetCache.erase_if([&](TxSetFramCacheItem const& i) {
```

Comment:
I don't think this is a clean data source:
* txset cache is a cache -> there is no guarantee that you'll find all entries
* as we're counting when slots are evicted, there is a higher chance that eviction took place
* more noise may be added if multiple slots are purged at once

There is a function `logQuorumInformation` that seems to be a better place for this kind of tracking: it's only called when externalizing the latest slot, and with a ~fixed window to allow for messages to propagate.
I _think_ that you could combine something like `processCurrentState` and `getTxSetHashes` to build a set of unique values referenced by that slot.

Reply:
Yeah, I think the initial idea was to get an estimate on how many values the node is processing per ledger, which included values we pulled but didn't record (if an envelope got rejected, for example). I agree that looking at the cache can be noisy, though in this case eviction is extremely unlikely (the cache is sufficiently big, as I'm assuming we want to be able to answer queries about recent ledgers). That being said, I think a metric for values propagated network-wide makes more sense here. I updated the code based on your suggestion.